mp4san/src/parse/stsz.rs

```diff
@@ -0,0 +1,86 @@
+#![allow(missing_docs)]
+
+use std::num::NonZeroU32;
+
+use mediasan_common::{Report, ResultExt as _};
+
+use super::error::ParseResultExt as _;
+use super::{BoundedArray, BoxType, ConstFullBoxHeader, ConstU32, Mp4Value, ParseBox, ParseError, ParsedBox};
+use crate::error::Result;
+
+#[derive(Clone, Debug, ParsedBox)]
+#[box_type = "stsz"]
+pub enum StszBox {
+    FixedSize {
+        _parsed_header: ConstFullBoxHeader,
+        size: NonZeroU32,
+        number_of_samples: u32,
+    },
+    VariableSize {
+        _parsed_header: ConstFullBoxHeader,
```

Comment:
Hmm, another idea.. it might be nice to have support in the proc macro for something like:

```rust
#[derive(Clone, Debug, ParseBox, ParsedBox)]
#[box_type = "stsz"]
pub enum StszBox {
    FixedSize {
        #[parse_box_backtrack]
        _parsed_header: ConstFullBoxHeader,
        size: NonZeroU32,
        number_of_samples: u32,
    },
    VariableSize {
        #[parse_box_backtrack]
        _parsed_header: ConstFullBoxHeader,
        _not_fixed_size: ConstU32,
        entries: BoundedArray<u32, u32>,
    },
}
```

For lack of a better name, `#[parse_box_backtrack]` would tell the `ParseBox` derive to allow backtracking during parsing across that field, allowing `size`/`_not_fixed_size` to be the fields whose success decides which enum variant to continue parsing..

Reply:
At some point I'm reluctant to add features to proc-macros that can be implemented with normal language features (wrapper types).